pkg/api/service/annotations.go

```diff
@@ -25,4 +32,58 @@ const (
 	//
 	// Not all cloud providers support this annotation, though AWS & GCE do.
 	AnnotationLoadBalancerSourceRangesKey = "service.beta.kubernetes.io/load-balancer-source-ranges"
+
+	// AnnotationExternalTraffic An annotation that denotes if this Service desires to route external traffic to local
+	// endpoints only. This preserves Source IP and avoids a second hop.
+	AnnotationExternalTraffic = "service.alpha.kubernetes.io/external-traffic"
+	// AnnotationValueExternalTrafficLocal Value of annotation to specify local endpoints behaviour
+	AnnotationValueExternalTrafficLocal = "OnlyLocal"
+	// AnnotationValueExternalTrafficGlobal Value of annotation to specify global (legacy) behaviour
+	AnnotationValueExternalTrafficGlobal = "Global"
+	// AnnotationHealthCheckNodePort Annotation specifying the healthcheck nodePort for the service
+	// If not specified, annotation is created by the service api backend with the allocated nodePort
+	// Will use user-specified nodePort value if specified by the client
+	AnnotationHealthCheckNodePort = "service.alpha.kubernetes.io/healthcheck-nodeport"
```

Comment:
This allocates a max f one NodePort per service, but yeah.  Any experts can
point in the right direction?

On Sep 5, 2016 11:58 AM, "Clayton Coleman" <notifications@github.com> wrote:

> In pkg/api/service/annotations.go
> <https://github.com/kubernetes/kubernetes/pull/29409#discussion_r77550810>
> :
>
> > @@ -25,4 +32,58 @@ const (
> >  	//
> >  	// Not all cloud providers support this annotation, though AWS & GCE do.
> >  	AnnotationLoadBalancerSourceRangesKey = "service.beta.kubernetes.io/load-balancer-source-ranges"
> > +
> > +	// AnnotationExternalTraffic An annotation that denotes if this Service desires to route external traffic to local
> > +	// endpoints only. This preserves Source IP and avoids a second hop.
> > +	AnnotationExternalTraffic = "service.alpha.kubernetes.io/external-traffic"
> > +	// AnnotationValueExternalTrafficLocal Value of annotation to specify local endpoints behaviour
> > +	AnnotationValueExternalTrafficLocal = "OnlyLocal"
> > +	// AnnotationValueExternalTrafficGlobal Value of annotation to specify global (legacy) behaviour
> > +	AnnotationValueExternalTrafficGlobal = "Global"
> > +	// AnnotationHealthCheckNodePort Annotation specifying the healthcheck nodePort for the service
> > +	// If not specified, annotation is created by the service api backend with the allocated nodePort
> > +	// Will use user-specified nodePort value if specified by the client
> > +	AnnotationHealthCheckNodePort = "service.alpha.kubernetes.io/healthcheck-nodeport"
>
> Does this let a client request a new NodePort? If so this needs to be
> quota protected (like nodePorts) because they are a finite resource and
> this can allow exhaustion attacks.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/29409/files/b82c028f77d85ac0be39b510ccdbc2783e43fdc7#r77550810>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFVgVLYbJbaO9EXrh0N_L1yuemTvxBblks5qnGZwgaJpZM4JSUXH>
> .
>